### DISCUSSION TITLE
The argument type 'AlwaysAliveProviderBase<Stream<dynamic>>' can't be assigned to the parameter type 'ProviderListenable<Stream<bool>>

### DISCUSSION THREAD

**Discussion by UsetIT** [rrousselGit/riverpod#1073] [Q&A]
**Describe the bug**
I'm doing everything just like the docs say but I get this problem. 

**To Reproduce**

```
Widget build(BuildContext context, ref) {
    final String languageCode = 'ar';
    return MaterialApp(
      title: 'The Savior TV Channel',
      debugShowCheckedModeBanner: false,
      home: StreamBuilder<bool>(
          stream: ref.watch(internetStream.stream), // <-- Here,
          builder: (context, AsyncSnapshot<bool> snapshot) {
            print('data +' + snapshot.data.toString());
            return Splash();
          }),
```

and this is my StreamProvider:-

```
final internetStream = StreamProvider.autoDispose((_) {
  Stream<bool>? stream;

  SimpleConnectionChecker _simpleConnectionChecker = SimpleConnectionChecker()
    ..setLookUpAddress('pub.dev'); //Optional method to pass the lookup string
  stream = _simpleConnectionChecker.onConnectionChange;

  return stream;
});


```

**Expected behavior**
It should work just like the docs say here:- https://riverpod.dev/docs/concepts/reading#deciding-what-to-read

But I get this error:- 

: Error: The argument type 'AlwaysAliveProviderBase<Stream<dynamic>>' can't be assigned to the parameter type 'ProviderListenable<Stream<bool>>'.
lib/main.dart:20
- 'AlwaysAliveProviderBase' is from 'package:riverpod/src/framework.dart' ('/F:/Flutter/flutter/.pub-cache/hosted/pub.dartlang.org/riverpod-1.0.3/lib/src/framework.dart').
package:riverpod/src/framework.dart:1
 - 'Stream' is from 'dart:async'.
- 'ProviderListenable' is from 'package:riverpod/src/framework.dart' ('/F:/Flutter/flutter/.pub-cache/hosted/pub.dartlang.org/riverpod-1.0.3/lib/src/framework.dart').
package:riverpod/src/framework.dart:1
    Stream<bool> internet = ref.watch(internetStream.stream);

Apologies if I'm being a newbie!

**Comment by rrousselGit** [comment]
You're missing the generic type on your provider:

```diff
-final internetStream = StreamProvider.autoDispose((_) {
+final internetStream = StreamProvider.autoDispose<bool>((_) {
  Stream<bool>? stream;

  SimpleConnectionChecker _simpleConnectionChecker = SimpleConnectionChecker()
    ..setLookUpAddress('pub.dev'); //Optional method to pass the lookup string
  stream = _simpleConnectionChecker.onConnectionChange;

  return stream;
});
```
